sys/trickle/trickle.c

```diff
@@ -30,63 +27,61 @@ void trickle_callback(trickle_t *trickle)
     if ((trickle->c < trickle->k) || (trickle->k == 0)) {
         (*trickle->callback.func)(trickle->callback.args);
     }
+
+    trickle_interval(trickle);
 }
 
 void trickle_interval(trickle_t *trickle)
 {
-    uint32_t max_interval;
+    uint32_t old_interval = trickle->I;
```

Comment:
but the [RFC](https://tools.ietf.org/html/rfc6206#section-4.2) states:

```
2.    When an interval begins, Trickle resets c to 0 and sets t to a
       random point in the interval, taken from the range [I/2, I), that
       is, values greater than or equal to I/2 and less than I.  The
       interval ends at I.
```
 and 
```
 5.  When the interval I expires, Trickle doubles the interval length.
       If this new interval length would be longer than the time
       specified by Imax, Trickle sets the interval length I to be the
       time specified by Imax.
```

It doesn't say anything about `remember the old value of I` - IMHO the code here is wrong!